### PR TITLE
Fix cross-dyadic product: Ensure minor- and major-symmetric result

### LINKED ISSUE #37: Inner-crossed dyadic product of two sym. tensors can't be stored as a 6x6 fourth-order tensor
The result of the inner crossed-dyadic product of two symmetric tensors $\boldsymbol{A}$ and $\boldsymbol{B}$

$$C_{ijkl} = A_{ik} \cdot B{jl} $$

is unsymmetric and hence, can't be stored as a 6x6 tensor. 

$$C_{ijkl} \ne C_{jikl} \qquad \text{and} \qquad C_{ijkl} \ne C_{ijlk} $$

The implementation is wrong.

### LINKED ISSUE #39: Rebase hardcoded fourth-order `identity4(T)` tensor on `cdya(T, T)`
as these functions are equal.

### LINKED ISSUE #40: Crossed-dyadic product is wrong
as already pointed out in #37 ,

the result of the crossed-dyadic product of two symmetric tensors $\boldsymbol{A}$ and $\boldsymbol{B}$

$$C_{ijkl} = \frac{1}{2} \left(A_{ik} \cdot B_{jl} +A_{il} \cdot B_{jk} \right) $$

is not symmetric and hence, can't be stored as a 6x6 tensor. 

The implementation is wrong. The current implementation is only valid if $\boldsymbol{A} = \boldsymbol{B}$!

### PR DESCRIPTION
**Pull request by adtzlr**
- [x] make the cross-dyadic product fully symmetric, i.e. 
        `C(i,j,k,l) = (A(i,k) B(j,l) + A(i,l) B(j,k) + B(i,k) A(j,l) + B(i,l) A(j,k))/4`
        closes #40
        see also [Maria Holland's Hitchhikers Guide to Abaqus](https://github.com/mholla/hitchhikers-guide-to-abaqus/blob/master/documentation.pdf), p.46. Btw, thank you @mholla for this great summary on symmetry, I've overlooked that over all these years!

- [x] rebase `identity4(T)` on the fixed `cdya(T, T)`
        closes #39 

- [x] remove the incorrect implementation of `the inner cross-dyadic icdya(T, T)`
        closes #37 

Especially the first point was hard to detect because I've used the special case `A = B` in 99% of all applications where the implementation up to [`v1.1.2`](https://github.com/adtzlr/ttb/releases/tag/v1.1.2) works without any problems.

### CODE REVIEW

**Comment by adtzlr** [comment]
@jfriedlein I did my best in fixing it with gfortran because currently I have no FEA code / intel fortran at hand to test the changes. Could you check if your models converge with these changes?

Many many thanks in advance 😄

File: docs/index.md [hunk 1]
```diff
@@ -37,7 +36,10 @@ Andreas Dutzler. *Tensor Toolbox for Modern Fortran - High-Level Tensor Manipula
...
 I'm a PhD student at the Insitute of Structural Durability and Railway Technology at Graz University of Technology.
 
+# Changelog
+All notable changes to this project will be documented in [this file](https://github.com/adtzlr/ttb/blob/main/CHANGELOG.md). The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
 ## Credits
 Naumann, C.: [Chemisch-mechanisch gekoppelte Modellierung und Simulation oxidativer Alterungsvorgänge in Gummibauteilen (German)](http://nbn-resolving.de/urn:nbn:de:bsz:ch1-qucosa-222075). PhD thesis. Fakultät für Maschinenbau der Technischen Universität Chemnitz, 2016.
 
-## Last updated: 2021/08/30
+## Last updated: 2023/07/37
```
Comment (jfriedlein) [inline docs/index.md:45]: 2023/07/07

Comment (adtzlr) [reply docs/index.md:45]: Greetings from the universe 🚀 - thanks!

**Review by jfriedlein — COMMENTED** [review]
Besides the small typo with the date, it looks fine.

**Comment by adtzlr** [comment]
Perfect! I checked the numeric results with my Python package [adtzlr/hyperelastic](https://hyperelastic.readthedocs.io/en/latest/hyperelastic/global/math.html#hyperelastic.math.cdya) but I couldn't test an actual UMAT / HYPELA2 / whatever subroutine inside a FEA package if something happened related to convergence.

**Comment by jfriedlein** [comment]
Hi @adtzlr,

I tested the "fix-crossdyadic" branch for LS-Dyna:
- Ubuntu 20.04.6 LTS, gfortran GNU Fortran (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
- Windows 10, Visual Studio 2017, ifort17
For both it compiled fine. Also the tensor functions that I use, worked as perfect as before. (I noticed something that is unrelated to the current fix, so I added this as issue #45).

For a more reliable and comprehensive test, we should think about unit tests. Probably that is a lot of work to set up, but once the framework is ready, it should be easy to add more and more tests and provide a quality assurance. Maybe a todo for the future.

**Comment by adtzlr** [comment]
Yes, unit tests are indeed a great idea! Ideally would be an automated CI/CD pipeline with Github Actions. But my experience in Fortran ends with the scope of this package... I usually do some UMAT scripting but thats it.

File: docs/index.md (hunk 1 above)
Comment (adtzlr) [reply docs/index.md]: Resolved in [cdcdf9e](https://github.com/adtzlr/ttb/pull/44/commits/cdcdf9e6672a58716c73f7761ba663b9ea9a37d4).

**Comment by mholla** [comment]
> * see also [Maria Holland's Hitchhikers Guide to Abaqus](https://github.com/mholla/hitchhikers-guide-to-abaqus/blob/master/documentation.pdf), p.46. Btw, thank you @mholla for this great summary on symmetry, I've overlooked that over all these years!

ah wow, very happy to hear that my document was helpful!

**Pull request merged**